test/logstream/v2/stream.go

```diff
@@ -34,19 +34,19 @@ import (
 	"knative.dev/pkg/ptr"
 )
 
-func FromNamespace(ctx context.Context, c kubernetes.Interface, namespace string) Source {
+func FromNamespace(ctx context.Context, c kubernetes.Interface, namespaces []string) Source {
 	return &namespaceSource{
-		ctx:       ctx,
-		kc:        c,
-		namespace: namespace,
-		keys:      make(map[string]Callback, 1),
+		ctx:        ctx,
+		kc:         c,
+		namespaces: namespaces,
+		keys:       make(map[string]Callback, 1),
 	}
 }
 
 type namespaceSource struct {
```

Comment:
ditto,
which also makes me thing may be we want a collection of per-namespace objects instead

Reply:
IT would be a natural extension of this structure, but I think multi namespace collection (as it is today) will be used as one off  CLI controlled through environment variable with a single key.(so it is easy to get unfiltered logs from queue proxy, istio etc) , I do not really see immediate need to support multiple keys in that scenario